app/bundles/LeadBundle/Command/ContactScheduledExportCommand.php

```diff
@@ -55,17 +63,23 @@ protected function execute(InputInterface $input, OutputInterface $output): int
 
         $count = 0;
 
-        foreach ($contactExportSchedulers as $contactExportScheduler) {
-            $contactExportSchedulerEvent = new ContactExportSchedulerEvent($contactExportScheduler);
-            $this->eventDispatcher->dispatch($contactExportSchedulerEvent, LeadEvents::CONTACT_EXPORT_PREPARE_FILE);
-            $this->eventDispatcher->dispatch($contactExportSchedulerEvent, LeadEvents::CONTACT_EXPORT_SEND_EMAIL);
-            $this->eventDispatcher->dispatch($contactExportSchedulerEvent, LeadEvents::POST_CONTACT_EXPORT_SEND_EMAIL);
-            ++$count;
-        }
+        try {
+            foreach ($contactExportSchedulers as $contactExportScheduler) {
+                $contactExportSchedulerEvent = new ContactExportSchedulerEvent($contactExportScheduler);
+                $this->eventDispatcher->dispatch($contactExportSchedulerEvent, LeadEvents::CONTACT_EXPORT_PREPARE_FILE);
+                $this->eventDispatcher->dispatch($contactExportSchedulerEvent, LeadEvents::CONTACT_EXPORT_SEND_EMAIL);
+                $this->eventDispatcher->dispatch($contactExportSchedulerEvent, LeadEvents::POST_CONTACT_EXPORT_SEND_EMAIL);
+                ++$count;
+            }
 
-        $output->writeln('Contact export email(s) sent: '.$count);
+            $output->writeln('Contact export email(s) sent: '.$count);
 
-        return Command::SUCCESS;
+            return ExitCode::SUCCESS;
+        } catch (SignalCaughtException) {
+            $output->writeln('touch job');
```

Comment:
This doesn't make sense for the community and I don't think we need this in ACS either as the job is being terminated anyway, right?

Reply:
@escopecz `return ExitCode::TERMINATED;` **is crucial** here as it communicates with the parent process. It tells the parent process that the child process **was gracefully terminated**.

Reply:
@escopecz if you meant just `$output->writeln('touch job');`, yes, it is unnecessary to both Community and ACS.

Reply:
Yes, I meant the touch job line